lib/sinatra/base.rb

```diff
@@ -918,15 +914,14 @@ def call!(env) # :nodoc:
       @response = Response.new
       template_cache.clear if settings.reload_templates
 
-      @response['Content-Type'] = nil
       invoke { dispatch! }
       invoke { error_block!(response.status) } unless @env['sinatra.error']
 
       unless @response['Content-Type']
-        if Array === body and body[0].respond_to? :content_type
+        if Array === body && body[0].respond_to?(:content_type)
           content_type body[0].content_type
```

Comment:
Is this code path ever used? I searched, but haven't found any usage example or test for it.
Could someone explain to me why do we need it?